src/plugins/lua/replies.lua

```diff
@@ -91,16 +97,146 @@ local function replies_check(task)
           in_reply_to, stored_rcpt)
     end
 
-    return false
+    return nil
+  end
+
+  local function configure_redis_scripts(recipients, task_time_str)
+    local redis_script_zadd_global = [[
+      local global_size = redis.call('ZCARD', KEYS[1])
+      if global_size > {= mgs =} then
+        redis.call('ZREMRANGEBYRANK', KEYS[1], '{= mgs =}', global_size)
+      end
+      -- adding recipients to the global replies set
+      redis.call('ZADD', KEYS[1], ARGV[1], ARGV[2])
+      ]]
+    local set_script_zadd_global = lua_util.jinja_template(redis_script_zadd_global,
+              { mgs = settings['max_global_size'] })
+    table.insert(script_ids, lua_redis.add_redis_script(set_script_zadd_global, redis_params))
+
+    local redis_script_zadd_local = [[
+      local local_replies_set_size = redis.call('ZCARD', KEYS[1])
+      if local_replies_set_size > {= mls =} then
+        redis.call('ZREMRANGEBYRANK', KEYS[1], '{= mls =}', local_replies_set_size)
+      end
+      -- adding recipients for the local replies set
+      redis.call('ZADD', KEYS[1], ARGV[1], ARGV[2])
+      -- setting expire for local replies set
+      redis.call('EXPIRE', KEYS[1], tostring(math.floor('{= exp =}')))
+    ]]
+    local set_script_zadd_local = lua_util.jinja_template(redis_script_zadd_local,
+      { exp = settings['expire'], mls = settings['max_local_size'] })
+    table.insert(script_ids, lua_redis.add_redis_script(set_script_zadd_local, redis_params))
```

Comment:
I'd recommend to have `script_ids` being indexed not by a numeric index but by some reasonable string index.